orc8r/tools/ansible/roles/pkgrepo/tasks/main.yml

```diff
@@ -10,28 +10,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# TODO move to preburn
+- name: Collect all registries
+  ansible.builtin.find:
+    paths: /etc/apt/sources.list.d/
+    patterns: "*.list*"
+  register: registries_to_be_delete
+  when: full_provision
+
 # TODO move to preburn
 - name: Remove all old registries
-  become: yes
-  ignore_errors: yes
   ansible.builtin.file:
-    path: /etc/apt/sources.list.d/*.list
+    path: "{{ item.path }}"
     state: absent
-  when: full_provision
+  with_items: "{{ registries_to_be_delete.files }}"
 
 # TODO move to preburn
 - name: Adding the key for the registry
-  apt_key:
-    url: https://linuxfoundation.jfrog.io/artifactory/api/security/keypair/magmaci/public
-    state: present
+  ansible.builtin.copy:
+    src: /home/vagrant/magma/keys/linux_foundation_registry_key.gpg
```

Comment:
In `lte/gateway/deploy/roles/magma_deploy/tasks/main.yml` we take the key from the Ansible files, but here we reference them via absolute path to the mounted repository.

What we are doing here only works because we happen to have the repository mounted at `/home/vagrant/magma` while provisioning. This comment is a bit nitpicky as there are already plenty of occurrences of this pattern, but I think it makes sense to move slowly towards the cleaner pattern of taking the key from the Ansible files. Especially since we already do that for `magma_deploy/tasks/main.yml`

Reply:
Yes, this was intentional (to reduce the number of checked in keys). Would you see it as best practice to have the corresponding files to an Ansible step in the respective `files` folder?
Is there a similar best practice for the Dockerfiles? In the PR, all Dockerfiles get the GPG key from the centralized location.

Reply:
> Would you see it as best practice to have the corresponding files to an Ansible step in the respective files folder?

Yes, I would say this is the clean solution. I don't have a elaborate idea for deduplicating the keys files though. Maybe one could use symlinks, but I guess that doesn't work. Probably the cleanest would be to have one Ansible role that deploys the key, and use that Ansible role for all playbooks that need it. I think this is a pattern that we already use, at least [this line](https://github.com/magma/magma/blob/3e0b75ba6ac5391ad9034f507bd8251bd3deb9e8/lte/gateway/ansible.cfg#L7) indicates that we are using roles defined in the `orc8r` folder for the LTE AGW.

That said, while looking this up I found that we have this hardcoded `/home/vagrant/magma` so heavily baked into our Ansible files that I'm not sure if doing it in clean way is worth the extra effort after all :/ But in case we don't want the clean solution, we should at least consistently use the dirty solution ;)

Reply:
> In the PR, all Dockerfiles get the GPG key from the centralized location.

I think that's ok. It doesn't require the repo to be mounted at `/home/vagrant/magma`. Also, we are just COPYing the file from the Docker build context which is the standard thing to do.

Reply:
All Ansible roles now use their own key in their `files` folder. Docker and GitHub actions use the centralized key.